modules/cms-kit/src/Volo.CmsKit.Public.Web/Pages/CmsKit/Shared/Components/Rating/RatingViewComponent.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Volo.Abp.AspNetCore.Mvc;
+using Volo.Abp.AspNetCore.Mvc.UI;
+using Volo.Abp.AspNetCore.Mvc.UI.Widgets;
+using Volo.Abp.Users;
+using Volo.CmsKit.Public.Ratings;
+
+namespace Volo.CmsKit.Public.Web.Pages.CmsKit.Shared.Components.Rating
+{
+    [ViewComponent(Name = "CmsRating")]
+    [Widget(
+        StyleTypes = new[] {typeof(RatingStyleBundleContributor)},
+        ScriptTypes = new[] {typeof(RatingScriptBundleContributor)},
+        RefreshUrl = "/CmsKitPublicWidgets/Rating"
+    )]
+    public class RatingViewComponent : AbpViewComponent
+    {
+        public IRatingPublicAppService RatingPublicAppService { get; }
+        public AbpMvcUiOptions AbpMvcUiOptions { get; }
+        public ICurrentUser CurrentUser { get; }
+
+        public RatingViewComponent(IRatingPublicAppService ratingPublicAppService, IOptions<AbpMvcUiOptions> options, ICurrentUser currentUser)
+        {
+            RatingPublicAppService = ratingPublicAppService;
+            AbpMvcUiOptions = options.Value;
+            CurrentUser = currentUser;
+        }
+        
+        public virtual async Task<IViewComponentResult> InvokeAsync(string entityType, string entityId)
+        {
+            var ratings = await RatingPublicAppService.GetGroupedStarCountsAsync(entityType, entityId);

```

Comment:
For the reaction system, I've created a single method (IReactionPublicAppService.GetForSelectionAsync) that returns all the info to render the widget.

Making a second call here (RatingPublicAppService.GetCurrentUserRatingAsync) which decreases performance especially in a tiered environment. An app service may make multiple db calls, no problem, but a UI element should do app service call as minimum as possible (because it may be a REST call in the real life).